### PR TITLE
bindings n genesis refactor

### PR DESCRIPTION
**Pull request by macwis**
# Goals of PR

Core changes:

- Refactored bindings generation
- Refactored genesis generation and imports
- Refactored sidecar imports

Notes:

n/a

### CODE REVIEW

**Review by siosw — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -20,26 +20,22 @@ OPS_BIN_TARGET = ./build/bin/genesis # relative to OPS_DIR
...
 # TODO add clef back in when moving to services/el_clients/go-ethereum
 #CLEF_SRC = $(SIDECAR_DIR)/cmd/clef/
 #CLEF_TARGET = $(SIDECAR_BIN)/clef
 
 install: geth magi sidecar ops
-geth: ops-bindings $(GETH_BIN_TARGET)
+geth: bindings $(GETH_BIN_TARGET)
 magi: $(MAGI_BIN_TARGET)
-sidecar: bindings $(shell find $(SIDECAR_DIR) -type f -name "*.go")
+sidecar: $(shell find $(SIDECAR_DIR) -type f -name "*.go")
```
Comment (LEAFERx) [inline Makefile]: it still needs the bindings right?

File: bindings-go/Makefile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+CONTRACTS_DIR="${MONOREPO_BASE}/contracts"
+GETH_DIR := "${MONOREPO_BASE}/services/el_clients/go-ethereum"
+
+.PHONY: all # Run all necessary steps and generate bindings
+all: bindings
+
+.PHONY: abigen # Install latest abigen command binary
+abigen:
+	cd "${GETH_DIR}" && \
+		go run build/ci.go install ./cmd/abigen
```
Comment (LEAFERx) [inline bindings-go/Makefile:20]: where will this be installed?

File: bindings-go/artifacts.json [hunk 3]
```diff
@@ -6,5 +6,10 @@
     "src/pre-deploy/UUPSPlaceholder.sol",
     "src/pre-deploy/L2BaseFeeVault.sol",
     "src/pre-deploy/L1FeeVault.sol",
-    "src/pre-deploy/MintableERC20Factory.sol"
+    "src/pre-deploy/MintableERC20Factory.sol",
+
```
Comment (LEAFERx) [inline bindings-go/artifacts.json]: remove unnecessary empty line

**Review by LEAFERx — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (macwis) [reply Makefile]: yes, I overlooked this change from the earlier PR - good spot

File: bindings-go/Makefile (hunk 2 above)
Comment (macwis) [reply bindings-go/Makefile:20]: initially I wanted to do this by a download, but I run into issues, so I left this same as earlier in the /ops - just a compilation from our go-ethereum fork; it's not installed, just used later as arg in `-abigen-bin "${GETH_DIR}/build/bin/abigen"` of the `bindings` action

**Pull request merged**